mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -105,6 +106,55 @@ def test_io_set_raw(fname):
         assert_array_equal(raw0.annotations.duration, 0.0)
 
 
+@testing.requires_testing_data
+def test_io_set_preload_false_uses_lazy_loading():
+    """Ensure reading .set files without preload keeps data out of memory."""
+    raw_lazy = read_raw_eeglab(raw_fname_onefile_mat, preload=False)
+    raw_eager = read_raw_eeglab(raw_fname_onefile_mat, preload=True)
+
+    try:
+        assert not raw_lazy.preload
+        assert raw_lazy._data is None
+        assert raw_lazy.n_times == raw_eager.n_times
+        assert raw_lazy.info["nchan"] == raw_eager.info["nchan"]
+
+        lazy_slice = raw_lazy[:2, :10][0]
+        assert lazy_slice.shape == (2, 10)
+
+        # on-demand reads must not flip the preload flag or populate _data
+        assert not raw_lazy.preload
+        assert raw_lazy._data is None
+        assert raw_eager._data.nbytes > lazy_slice.nbytes
+    finally:
+        raw_lazy.close()
+        raw_eager.close()
```

Comment:
It's odd that you need to `.close` these... we don't do this anywhere else in our tests